src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetCache.cs

```diff
@@ -178,83 +220,158 @@ public static IEnumerable<T> EnumerateAllComponentsInScenesAndPrefabs<T>()
         }
     }
 
-    internal abstract class AssetCache<TAssetEntry, TAsset> : AssetCache, IAssetCache
-        where TAssetEntry : AssetCacheEntry<TAsset>, new()
+    internal abstract class AssetCache<TAsset> : AssetCache, IAssetCache
         where TAsset : UnityEngine.Object
     {
-        [SerializeField]
-        private TAssetEntry[] assets = null;
+        const string assetContentExtension = ".asset";
 
-        private Dictionary<AssetId, TAssetEntry> lookupByAssetId;
-        private Dictionary<TAsset, TAssetEntry> lookupByAsset;
+#pragma warning disable 414

```

Comment:
>414 [](start = 24, length = 3)

Non-nit: every time we disable a warning, we should have a comment about why the warning is being ignored/disabled (i.e. why it's safe to ignore)

Reply:
We have to set a good example for the future where people may copy-paste things willy-nilly, and commentless-warning disables can proliferate inadvertently this way.

---
In reply to: [355037489](https://github.com/microsoft/MixedReality-SpectatorView/pull/295#discussion_r355037489) [](ancestors = 355037489)